src/components/convex-hull-calculator.tsx

```diff
@@ -1,175 +1,288 @@
+    setPoints(points.map((p) => (p.id === editingPoint ? { ...p, x, y } : p)));
 
-    setEditingPoint(null)
-    setNewX("")
-    setNewY("")
-    setError(null)
-  }
+    setEditingPoint(null);
+    setNewX("");
+    setNewY("");
+    setError(null);
+  };
 
   const cancelEdit = () => {
-    setEditingPoint(null)
-    setNewX("")
-    setNewY("")
-    setError(null)
-  }
+    setEditingPoint(null);
+    setNewX("");
+    setNewY("");
+    setError(null);
+  };
 
   const deletePoint = (id: string) => {
-    setPoints(points.filter((p) => p.id !== id))
-  }
+    setPoints(points.filter((p) => p.id !== id));
+  };
 
   const drawCanvas = () => {
-    const canvas = canvasRef.current
-    if (!canvas) return
+    const canvas = canvasRef.current;
+    if (!canvas) return;
 
-    const ctx = canvas.getContext("2d")
-    if (!ctx) return
+    const ctx = canvas.getContext("2d");
+    if (!ctx) return;
 
     // Responsywność canvasu
-    const container = canvas.parentElement
+    const container = canvas.parentElement;
     if (container) {
-      canvas.width = container.clientWidth
-      canvas.height = Math.min(400, window.innerHeight * 0.5)
+      canvas.width = 600;
+      canvas.height = 600;
     }
 
-    // Czyszczenie canvasu
-    ctx.clearRect(0, 0, canvas.width, canvas.height)
+    // Tutaj sobie rysujemy na canvasie siatkę XY
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
+    ctx.strokeStyle = "#e0e0e0";
+    ctx.lineWidth = 1;
+    ctx.beginPath();
+    for (let i = 0; i <= canvas.width; i += 20) {
+      ctx.moveTo(i, 0);
+      ctx.lineTo(i, canvas.height);
+    }
+    for (let i = 0; i <= canvas.height; i += 20) {
+      ctx.moveTo(0, i);
+      ctx.lineTo(canvas.width, i);
+    }
+    ctx.stroke();
+    ctx.strokeStyle = "#000000";
+    ctx.lineWidth = 2;
+    ctx.beginPath();
+    ctx.moveTo(0, canvas.height / 2);
+    ctx.lineTo(canvas.width, canvas.height / 2);
+    ctx.moveTo(canvas.width / 2, 0);
+    ctx.lineTo(canvas.width / 2, canvas.height);
+    ctx.stroke();
+    ctx.fillStyle = "#000000";
+    ctx.font = "12px Arial";
+    ctx.fillText("X", canvas.width - 20, canvas.height / 2 + 15);
+    ctx.fillText("Y", canvas.width / 2 + 5, 20);
+    ctx.fillText("0", canvas.width / 2 + 5, canvas.height / 2 + 15);
+
+    // A tutaj na siatce pozycje -200 i 200 dla X i Y
+    ctx.fillStyle = "#000000";
+    ctx.fillText("-200", 85, 325);
+    ctx.fillText("200", 485, 325);
+    ctx.fillText("200", 315, 124);
+    ctx.fillText("-200", 315, 484);
+
+    // A tutaj są linie pomocnicze do 200, -200 dla X i Y
+    ctx.strokeStyle = "#000000";
+    ctx.lineWidth = 1;
+    ctx.beginPath();
+    ctx.moveTo(100, 300);
+    ctx.lineTo(100, 310);
+
+    ctx.moveTo(500, 300);
+    ctx.lineTo(500, 310);
+
+    ctx.moveTo(300, 120);
+    ctx.lineTo(310, 120);
+
+    ctx.moveTo(300, 480);
+    ctx.lineTo(310, 480);
+    ctx.stroke();
 
     // Jestli nie ma współrzędnych punktów, to nic nie rysujemy
```

Comment:
```suggestion
    // Jeżeli nie ma punktów, to nic nie rysujemy
```
Tak wiem, to mój komentarz 😅